src/Protocol/ActivityPub/Processor.php

```diff
@@ -268,6 +247,62 @@ public static function createEvent($activity, $item)
 		Logger::log('Event '.$event_id.' was stored', Logger::DEBUG);
 	}
 
+	/**
+	 * Process the content
+	 *
+	 * @param array $activity Activity array
+	 * @param array $item
+	 * @return array
+	 * @throws \Exception
+	 */
+	private static function processContent($activity, $item)
+	{
+		$item['title'] = HTML::toBBCode($activity['name']);
+
+		if (!empty($activity['source'])) {
+			$item['body'] = $activity['source'];
+		} else {
+			$content = HTML::toBBCode($activity['content']);
+			$content = self::replaceEmojis($content, $activity['emojis']);
+			$content = self::convertMentions($content);
+
+			if (($item['thr-parent'] != $item['uri']) && ($item['gravity'] == GRAVITY_COMMENT)) {
+				$item_private = !in_array(0, $activity['item_receiver']);
+				$parent = Item::selectFirst(['id', 'private', 'author-link', 'alias'], ['uri' => $item['thr-parent']]);
+				if (!DBA::isResult($parent)) {
+					Logger::warning('Unknown parent item.', ['uri' => $item['thr-parent']]);
+					return false;
+				}
+				if ($item_private && !$parent['private']) {
+					Logger::warning('Item is private but the parent is not. Dropping.', ['item-uri' => $item['uri'], 'thr-parent' => $item['thr-parent']]);
+					return false;
```

Comment:
Please return an empty array (`[]`) instead (no type mismatching). And AFAIK `empty` is working with empty arrays too.